pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
+                    f"Allowed precision values: {PrecisionType.supported_types()}"
+                )
+            self._precision_flag = precision
+
+        if plugins:
+            for plugin in plugins:
+                if isinstance(plugin, Strategy) or isinstance(plugin, str) and plugin in self._existing_strategies_str:
+                    self._strategy_flag = plugin
+                    rank_zero_deprecation(
+                        f"Passing {plugin} `strategy` to the `plugins` flag in Trainer has been deprecated"
+                        f" in v1.5 and will be removed in v1.7. Use `Trainer(strategy={plugin})` instead."
+                    )
 
-        for plug in self.plugins:
-            if isinstance(plug, str) and plug in StrategyRegistry:
-                if strategy is None:
-                    strategy = StrategyRegistry.get(plug)
+                elif isinstance(plugin, PrecisionPlugin):
+                    self._precision_plugin_flag = plugin
+                elif isinstance(plugin, str) and plugin in self._supported_precision:
+                    self._precision_flag = plugin
+                elif isinstance(plugin, CheckpointIO):
+                    self.checkpoint_io = plugin
+                elif isinstance(plugin, ClusterEnvironment):
+                    self._cluster_environment_flag = plugin
                 else:
                     raise MisconfigurationException(
-                        "You can only specify one precision and one training type plugin."
-                        " Found more than 1 training type plugin:"
-                        f' {StrategyRegistry[plug]["strategy"]} registered to {plug}'
+                        f"Found invalid type for plugin {plugin}. Expected a precision or training type plugin."
                     )
-            if isinstance(plug, str):
-                # Reset the distributed type as the user has overridden training type
-                # via the plugins argument
-                self._strategy_type = None
-                self.set_distributed_mode(plug)
 
-            elif isinstance(plug, Strategy):
-                if strategy is None:
-                    strategy = plug
-
-                else:
+        # handle the case when the user passes in a strategy instance which has an accelerator, precision,
+        # checkpoint io or cluster env set up
+        # TODO: @awaelchli imporve the error messages below
+        if self._strategy_flag and isinstance(self._strategy_flag, Strategy):
+            if self._strategy_flag._accelerator:
+                if self._accelerator_flag:
                     raise MisconfigurationException(
-                        "You can only specify one training type plugin."
-                        f" Available: {type(strategy).__name__}, given: {type(plug).__name__}"
+                        "accelerator set through both strategy class and accelerator flag, choose one"
```

Comment:
We can improve all the error message in a separate PR, otherwise we need update a bunch of tests to match the error message, the PR will be even bigger